content/search.html

```diff
@@ -0,0 +1,119 @@
+---
+title: Search Results
+---
+
+<div id="main" class="span12" role="main">
+    <h1>Results</h1>
+</div>
+<div id="wrap" class="container" role="document">
+  <div id="responsive-menu"><button class="btn btn-primary" onclick="handleMenu()">Table of Contents</button></div>
+  <div id="content" class="row-fluid">
+    <div class="toc well span3 visible-toc">
+      <div style="display: flex; flex-direction: row; align-items: space-between; justify-content: space-between;">
+        <input type="text" id="search-input" placeholder="Search docs" /><br/>        
+        <button style="width: 40% margin: auto; margin-left: 5px;" class="btn-primary" onclick="handleSearch()">Search</button>
+      </div>
+      {% include "docs-toc.html" %}
+    </div>
+    <div id="main-docs" class="well span9 docs">
+      <h1>{{ title }}</h1>
+      {{ content | safe }}   
+    </div>
+    <div class="well span9 docs" id="search-results" style="display: none;">
+      <h3>Search Results</h3>
+      <div id="results-ul">
+
+      </div>
+    </div>
+  </div>
+</div>
+
+<script>
+    document.addEventListener("DOMContentLoaded", async () => {
+      //  Listen for enter event
+      const search = document.getElementById("search-input")
+      search.addEventListener("keyup", ({key}) => {
+        if (key === "Enter") {
+          handleSearch();
+        }
+      })
+      //  Get the search term from the url
+      const parsedSearchTerm = decodeURI(window.location.href.split("=")[1]);
+      const docsContainer = document.getElementById("main-docs");
+      docsContainer.style.display = "none";
+      const resultsContainer = document.getElementById("search-results");
+      resultsContainer.style.display = "block";
+      const res = await fetch(`${window.location.origin}/assets/index.json`);
+      const data = await res.json();
+      const keys = Object.keys(data);
+      let docs = [];
+      for(const key of keys) {
+        docs.push(data[key]);
+      }
+
+      const uniqueDocs = [...new Set(docs)];
+      
+      const filtered = uniqueDocs.filter((doc) => doc.title.includes(parsedSearchTerm) || doc.content.includes(parsedSearchTerm));
```

Comment:
It appears that this is just a substring search. Problem is that if I just search for `hey` I get `they` which isn't the what I was searching for.

Reply:
A better way to do fuzzy search than this might be to use a library like this: https://www.npmjs.com/package/fuzzy-search 

It introduces another dependency but it seems super lightweight.

Reply:
It's definitely worth taking a look at.

Reply:
I think the search is usable as is now though. So we can push it out and continue to improve it.